test/Microsoft.ML.TimeSeries.Tests/TimeSeriesDirectApi.cs

```diff
@@ -515,5 +547,144 @@ public void AnomalyDetectionWithSrCnn()
                 k += 1;
             }
         }
+
+        [Fact]
+        public void RootCauseLocalizationWithDT()
+        {
+            // Create an root cause localizatiom input list.
+            var rootCauseLocalizationData = new List<RootCauseLocalizationData>() { new RootCauseLocalizationData(new DateTime(), new Dictionary<string, Object>(), new List<MetricSlice>() { new MetricSlice(new DateTime(), new List<Microsoft.ML.TimeSeries.Point>()) }, AggregateType.Sum, _aggSymbol) };
+
+            var ml = new MLContext(1);
+            // Convert the list of root cause data to an IDataView object, which is consumable by ML.NET API.
+            var data = ml.Data.LoadFromEnumerable(rootCauseLocalizationData);
+
+            // Create pipeline to localize root cause by decision tree.
+            var pipeline = ml.Transforms.LocalizeRootCauseByDT(nameof(RootCauseLocalizationTransformedData.RootCause), nameof(RootCauseLocalizationData.Input));
+
+            // Fit the model.
+            var model = pipeline.Fit(data);
+
+            // Test path:  input list -> IDataView -> Enumerable of RootCauseLocalizationInputs.
+            var transformedData = model.Transform(data);
+
+            // Load input list in DataView back to Enumerable.
+            var transformedDataPoints = ml.Data.CreateEnumerable<RootCauseLocalizationTransformedData>(transformedData, false);
+
+            foreach (var dataPoint in transformedDataPoints)
+            {
+                var rootCause = dataPoint.RootCause;
+                Assert.NotNull(rootCause);
+            }
+
+            var engine = ml.Model.CreatePredictionEngine<RootCauseLocalizationData, RootCauseLocalizationTransformedData>(model);
+
+            DateTime timeStamp = GetCurrentTimestamp();
+            var newRootCauseInput = new RootCauseLocalizationData(timeStamp, GetAnomalyDimension(), new List<MetricSlice>() { new MetricSlice(timeStamp, GetRootCauseLocalizationPoints()) }, AggregateType.Sum, _aggSymbol);
+            var transformedRootCause = engine.Predict(newRootCauseInput);
+
+            Assert.NotNull(transformedRootCause);
+            Assert.Equal(1, (int)transformedRootCause.RootCause.Items.Count);
+
+            Dictionary<string, Object> expectedDim = new Dictionary<string, Object>();
+            expectedDim.Add("Country", "UK");
+            expectedDim.Add("DeviceType", _aggSymbol);
+            expectedDim.Add("DataCenter", "DC1");
+
+            foreach (KeyValuePair<string, object> pair in transformedRootCause.RootCause.Items[0].Dimension)
+            {
+                Assert.Equal(expectedDim[pair.Key], pair.Value);
+            }
+
+            var dummyData = ml.Data.LoadFromEnumerable(new List<String>() { "Test"});
+
+            //Create path
+            var modelPath = "temp.zip";
```

Comment:
>modelPath [](start = 16, length = 9)

Please change to something more specific to this test, to make sure that there are not other tests that might try to overwrite this.

Reply:
Also, use `DeleteOutputPath`, since it returns a complete path.

---
In reply to: [406675769](https://github.com/dotnet/machinelearning/pull/4925#discussion_r406675769) [](ancestors = 406675769)

Reply:
> Also, use `DeleteOutputPath`, since it returns a complete path.
> 
> In reply to: [406675769](https://github.com/dotnet/machinelearning/pull/4925#discussion_r406675769) [](ancestors = 406675769)

Have updated